servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/KeepAliveManagerTest.java

```diff
@@ -338,4 +396,33 @@ private static final class ScheduledTask {
             this.delayMillis = delayMillis;
         }
     }
+
+    private static final class KeepAliveManagerHandler extends ChannelInboundHandlerAdapter {
+        private KeepAliveManager keepAliveManager;
+
+        void keepAliveManager(KeepAliveManager keepAliveManager) {
+            this.keepAliveManager = requireNonNull(keepAliveManager);
+        }
+
+        @Override
+        public void channelInactive(final ChannelHandlerContext ctx) {
+            keepAliveManager.channelClosed();
```

Comment:
IIRC handlerRemoved and channelInactive can be both called one after another in the pipeline, is it necessary to add an "already closed" check in `KeepAliveManager#channelClosed`? The implementation of the method looks fine to me side-effect wise, just double checking with you if it's ok.

Reply:
Thanks for calling it out. It looks idempotent to me. It only overrides the states and cancels if there are any futures.